components/dashboard/src/components/IdeOptions.tsx

```diff
@@ -75,11 +85,20 @@ export const IdeOptionsModifyModal = ({
     onClose,
     updateMutation,
     ideOptions,
+    hidePinEditorInputs,
     ...props
 }: IdeOptionsModifyModalProps) => {
     const orgLevelEditorVersionPinningEnabled = useFeatureFlag("org_level_editor_version_pinning_enabled");
 
-    const ideOptionsArr = useMemo(() => (ideOptions ? sortedIdeOptions(ideOptions) : undefined), [ideOptions]);
+    const ideOptionsArr = useMemo(
+        () =>
+            ideOptions?.options
+                ? Object.values(ideOptions.options)
+                      .filter((x) => !x.hidden)
+                      .sort(IdeOptionsSorter)
+                : undefined,
+        [ideOptions],
+    );
```

Comment:
Can we move this logic to ide-options-query.ts we are still duplicating code in the end, I would say just move `sortedIdeOptions` and use that everywhere

Reply:
addressed